apps/prairielearn/src/sync/fromDisk/entity-list.ts

```diff
@@ -0,0 +1,113 @@
+interface NamedEntity {
+  name: string;
+}
+
+type ExistingEntity<Entity> = Entity & {
+  // TODO: make non-nullable once we make this non-null in the database schemas.
+  number: number | null;
+  implicit: boolean;
+};
+
+type ExtraEntity<Entity> = Entity & {
+  implicit: boolean;
+};
+
+type DesiredEntity<Entity> = Entity & { number: number; implicit: boolean };
+
+export function determineOperationsForEntities<Entity extends NamedEntity>({
```

Comment:
Maybe add some comments for this function? I think I understood everything by looking at a caller, but I struggled when reading this function in isolation.

Reply:
Done in 55c1b83, how do those look?